sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SerializationDiagnosticsContext.java

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import com.azure.cosmos.ZonedDateTimeSerializer;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+public class SerializationDiagnosticsContext {
+    public volatile List<SerializationDiagnosticsContext.SerializationDiagnostics> serializationDiagnosticsList;
+
+    public void addMetaDataDiagnostic(SerializationDiagnosticsContext.SerializationDiagnostics serializationDiagnostics) {
+        if (serializationDiagnosticsList == null) {
+            serializationDiagnosticsList = new ArrayList<>();
+        }
+
+        serializationDiagnosticsList.add(serializationDiagnostics);
+    }
+
+    public <T> T getResource(Callable<T> function, SerializationType serializationType) {
```

Comment:
also a general question, you changed the `deserialization()` -> `() -> deserialization()` everywhere thourhgout the code.

Any reason we are not capturing the start and end time of deserialization and pass here only the time window?

Reply:
This way we need not to put start,end time and adding it to the serialization list , that logic can be confined in one place.  There are some places where we can't use getResource() due to unavailability of serialization context  such as in getCreateDocumentRequest() in RxDocumentClientImpl.java , so for those we have addSerializationDiagnostics() as you suggesting.

Reply:
removed callable